stock_picking_customer_ref/models/stock_picking.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2015 AvanzOSC - Alfredo de la Fuente
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.depends('group_id', 'sale_id.client_order_ref')
+    def _compute_client_order_ref(self):
+        stock_pickings = self.filtered(
+            lambda stock_picking: stock_picking.group_id)
+        for picking in stock_pickings:
+            cond = [('procurement_group_id', '=', picking.group_id.id)]
+            sale = self.env['sale.order'].search(cond, limit=1)
```

Comment:
@bizzappdev The sale_id is now present in stock picking :

https://github.com/odoo/odoo/blob/12.0/addons/sale_stock/models/stock.py#L66

The traversal is direct through a related field.

Reply:
@bizzappdev Your changes are not enough. Please use related field instead.

```suggestion
            client_order_ref = fields.Char(
                string="Sale Reference/Description",
                related="sale_id.client_order_ref",
                store=True)
```

Reply:
Yes indeed it should be like this.